lib/phoenix_component.ex

```diff
@@ -720,8 +730,74 @@ defmodule Phoenix.Component do
     """)
   end
 
+  defp bad_type!(slot, name, type, line, file) do
+    compile_error!(line, file, """
+    invalid type #{inspect(type)} for attr #{inspect(name)} in slot #{inspect(slot)}. \
```

Comment:
Instead of duplicating the error message, considering doing something like this:

context = slot && " in slot #{inspect(slot)}"

and then the first line can be:

```suggestion
    invalid type #{inspect(type)} for attr #{inspect(name)}#{context}. \
```

Reply:
Actually, it seems the "attr FOO in slot BAR" is very common, so you can encapsulate it like this:

    defp attr_slot(attr, nil), do: "attr #{inspect(name)}"
    defp attr_slot(attr, slot), do: "attr #{inspect(name)} in slot #{inspect(slot)}"

And now you can remove the duplication across several messages in this module:

```suggestion
    invalid type #{inspect(type)} for #{attr_slot(name, slot)}. \
```